apps/dashboard/app/controllers/scripts_controller.rb

```diff
@@ -39,6 +45,8 @@ def submit
     opts = submit_script_params[:script].to_h.symbolize_keys
 
     if (job_id = @script.submit(opts))
+      @script.write_job_options_to_cache(opts)
```

Comment:
Can you move this into the `script#submit` function. I think it'd be better to encapsulate it there.